### PR TITLE
feature/utf8-db

### LINKED ISSUE #2519: Accented characters support for users fields
I am using an a google-based oauth authentication source and have enabled the automatic account creation for it. When the account is created and the first and last-names are imported into packetfence, if these users have accented characters in google, there seems to be an issue adding them to the local database.  Please see the image below for the imported user

image

This seems to be related to some utf-8 conversion being lost somewhere in the path. But it seems that the issue is more general with even manually created users having errors with accented characters. The user below was created manually on PF's creation screen.

image

Is there support for UTF-8 / non-ascii7 characters?

### LINKED ISSUE #4049: Use Unicode characters like Chinese/Arabic not are not correctly shown in web admin
Hi

I am using ZEN vm version. I get my users and authentication from my active directory and in the AD my users have Unicode characters.   It is not correctly shown in web admin and maybe in the database. Is there any solution to this problem now or in the future? 

Capture

### LINKED ISSUE #4514: v9-gui: display of UTF8 strings
Incorrect display of UTF8 strings in new GUI.

### Steps to reproduce

CentOS 7.6 running nightly builds

```console
# cat user_with_utf8.json
{"pid":"emilie","email":"emilie@foo.fr","firstname":"émilie"}

# file user_with_utf8.json
emilie_user.json: UTF-8 Unicode text

# cat user_with_utf8.json | pfperl-api get -M POST /api/v1/users | jq
{
  "id": "emilie",
  "message": "'emilie' created",
  "status": 201
}
```
### Actual results

In DB:
```
> select firstname from person where pid="emilie"\G;
*************************** 1. row ***************************
firstname: émilie
```

From API:
```console
# pfperl-api get /api/v1/user/emilie | jq .item.firstname
"Ã©milie"
```

In webadmin:
Screenshot from 2019-05-15 14-20-18

### Expected results

Correct display of strings in new GUI.

### Additional notes

On Debian 9, I got wrong encoding in DB:
```
MariaDB [pf]> select firstname from person where pid="emilie"\G;
*************************** 1. row ***************************
firstname: Ã©milie
```

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
UTF8 support in the database.

# Impacts
The database encoding.

# Issue
fixes #4514
fixes #2519
fixes #4049

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Full UTF8 support

## Bug Fixes
* Display of UTF8 strings (#4514)
* Accented characters support for users fields (#2519)
* Use Unicode characters like Chinese/Arabic not are not correctly shown in web admin (#4049)

### CODE REVIEW

**Comment by extrafu** [comment]
I think we should run the performance/stress tests we run prior a release on this PR before merging. See if it has any performance impacts for key aspects of PacketFence.

**Comment by extrafu** [comment]
@julsemaan I think we should look into this one and merge it sooner than later.

**Comment by julsemaan** [comment]
Agreed, IMO we need to:
- Rebase against the latest devel
- Get confirmation from @jrouzierinverse that he is getting consistent encoding across the captive portal + admin
- Get confirmation from @jrouzierinverse that he did test the resolution of the 3 issues he mentioned in this PR

File: db/upgrade-X.X-X.Y.sql [hunk 1]
```diff
@@ -51,6 +56,7 @@ SOURCE /usr/local/pf/db/upgrade-tenant-11.2-12.0.sql;
 
 DROP PROCEDURE IF EXISTS ValidateVersion;
 
+<<<<<<< HEAD
```
Comment (satkunas) [inline db/upgrade-X.X-X.Y.sql]: fix merge

File: db/upgrade-X.X-X.Y.sql [hunk 2]
```diff
@@ -534,12 +540,18 @@ BEGIN
...
+ALTER TABLE sms_carrier
+    CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_bin;
+>>>>>>> b9e33a202b (Upgrade database all tables, triggers, stored procedures to utf8mb4)
 
 INSERT INTO sms_carrier
     (name, email_pattern, created)
 VALUES
     ('RingRing', '%s@smsemail.be', now());
 
+<<<<<<< HEAD
```
Comment (satkunas) [inline db/upgrade-X.X-X.Y.sql]: fix merge

**Review by satkunas — CHANGES_REQUESTED** [review]

**Comment by extrafu** [comment]
@satkunas All good on this one?

**Comment by satkunas** [comment]
> @satkunas All good on this one?

I'm good

**Comment by extrafu** [comment]
@satkunas In this case please merge it

**Pull request merged**